MapboxDirections/Match/MBTracepoint.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+@objc(MBTracepoint)
+public class Tracepoint: Waypoint {
+    
+    /**
+     Number of probable alternative matchings for this trace point. A value of zero indicates that this point was matched unambiguously.
+     */
+    @objc open var alternateCount: Int
+    
+    /**
+     Index of the waypoint inside the matched route.
```

Comment:
It’s an antipattern to put the index of an item inside the item itself. As soon as you isolate the tracepoint from the array of tracepoints, this property becomes meaningless or misleading.

Instead, Match should have a `waypoints` property that contains whichever items in `tracepoints` happens to be a waypoint. If for some reason you want to know the index of a waypoint, you could look up the Tracepoint object’s index in `Match.waypoints`.

/ref https://github.com/mapbox/MapboxDirections.swift/pull/236#discussion_r163961821